Source/Solver/ChargeDensity.cpp

```diff
@@ -2,20 +2,46 @@
 
 // Compute rho in SC region for given phi
 void ComputeRho(MultiFab&      PoissonPhi,
+                Array<MultiFab, AMREX_SPACEDIM> &P_old,
                 MultiFab&      rho,
                 MultiFab&      e_den,
                 MultiFab&      p_den,
-		const MultiFab& MaterialMask)
+		const MultiFab& MaterialMask,
+                const Geometry& geom,
+                const amrex::GpuArray<amrex::Real, AMREX_SPACEDIM>& prob_lo,
+                const amrex::GpuArray<amrex::Real, AMREX_SPACEDIM>& prob_hi)
 {
+    // Calculate average Pr. We take the average only over the FE region
+    Real average_P_r = 0.;
+    Real total_P_r = 0.;
+    Real FE_index_counter = 0.;
+    
+    Compute_P_av(P_old, total_P_r, MaterialMask, FE_index_counter, average_P_r);
+
+    //Calculate integrated electrode charge (Qe) based on eq 13 of https://pubs.aip.org/aip/jap/article/44/8/3379/6486/Depolarization-fields-in-thin-ferroelectric-films
+    Real FE_thickness = FE_hi[2] - FE_lo[2];
+    Real coth = (exp(2.*metal_thickness/metal_screening_length) + 1.0) / (exp(2.*metal_thickness/metal_screening_length) - 1.0);
+    Real csch = (2.*exp(metal_thickness/metal_screening_length)) / (exp(2.*metal_thickness/metal_screening_length) - 1.0);
+    Real numerator = 0.5 * FE_thickness * average_P_r / epsilonX_fe;
+    Real denominator = metal_screening_length/epsilon_de*(coth - csch + FE_thickness/(2.*epsilonX_fe));
```

Comment:
```suggestion
    Real denominator = metal_screening_length/epsilon_de*(coth - csch) + FE_thickness/(2.*epsilonX_fe);
```
There's a typo in the paper